docs/content/develop/run-tests.md

```diff
@@ -1,53 +1,186 @@
 ---
-title: Use built provider
-weight: 40
+title: "Run tests"
+weight: 70
 aliases:
+  - /docs/getting-started/run-provider-tests
   - /docs/getting-started/use-built-provider
+  - /get-started/run-provider-tests
+  - /get-started/use-built-provider
+  - /getting-started/run-provider-tests
   - /getting-started/use-built-provider
 ---
 
-# Use built provider locally
+# Run tests
 
-{{< hint info >}}
-**Note:** If you want to test changes you've made in Magic Modules, you need to first [generate](/magic-modules/docs/getting-started/generate-providers/) the provider you want to test.
-{{< /hint >}}
+## Before you begin
+
+1. [Generate the modified provider(s)](/magic-modules/docs/getting-started/generate-providers/)
+
+
+1. Set up application default credentials for Terraform
+
+    ```bash
+    gcloud auth application-default login
+    export GOOGLE_USE_DEFAULT_CREDENTIALS=TRUE
+    ```
+
+1. Set the following environment variables:
+
+    ```bash
+    export GOOGLE_USE_DEFAULT_CREDENTIALS=TRUE
+    export GOOGLE_PROJECT=PROJECT_ID
+    export GOOGLE_REGION=us-central1
+    export GOOGLE_ZONE=us-central1-a
+    ```
+    Replace `PROJECT_ID` with the ID of the Google Cloud project you are using for testing.
+
+1. Optional: Some tests may require additional variables to be set, such as:
+
+    ```
+    GOOGLE_ORG
+    GOOGLE_BILLING_ACCOUNT
+    ```
+
+## Run automated tests
+
+{{< tabs "version" >}}
+
+{{< tab "GA Provider" >}}
+
+
+1. Run unit tests and linters
+
+    ```bash
+    cd $GOPATH/src/github.com/hashicorp/terraform-provider-google
+    make test
+    make lint
+    ```
+
+
+1. Run acceptance tests for only modified resources. (Full test runs can take over 9 hours.) See Go's documentation for more information about `-run`.
+
+    ```bash
+    make testacc TEST=./google TESTARGS='-run=TestAccContainerNodePool'
+    ```
+
+    TESTARGS allows you to pass [testing flags](https://pkg.go.dev/cmd/go#hdr-Testing_flags) to `go test`. The most important is `-run`, which allows you to limit the tests that get run. There are 2000+ tests, and running all of them takes over 9 hours and requires a lot of GCP quota.
+
+    `-run` is regexp-like, so multiple tests can be run in parallel by specifying a common substring of those tests (for example, `TestAccContainerNodePool` to run all node pool tests).
+
+1. Optional: Save verbose test output to a file for analysis.
+
+    ```bash
+    TF_LOG=TRACE make testacc TEST=./google TESTARGS='-run=TestAccContainerNodePool_basic' > output.log
+    ```
+
+1. Optional: debug tests with [Delve](https://github.com/go-delve/delve):
+
+    ```bash
+    ## Navigate to the google package within your local GCP Terraform provider Git clone.
+    cd $GOPATH/src/github.com/terraform-providers/terraform-provider-google/google
```

Comment:
This line shouldn't be necessary since they're already in the directory in this part of the guide.

```suggestion
```